HeterogeneousCore/AlpakaInterface/test/alpaka/testOneHistoContainer.dev.cc

```diff
@@ -0,0 +1,178 @@
+#include <algorithm>
+#include <iostream>
+#include <limits>
+#include <random>
+
+#include "HeterogeneousCore/AlpakaInterface/interface/config.h"
+#include "HeterogeneousCore/AlpakaInterface/interface/memory.h"
+#include "HeterogeneousCore/AlpakaInterface/interface/workdivision.h"
+#include "HeterogeneousCore/AlpakaInterface/interface/HistoContainer.h"
+
+using namespace cms::alpakatools;
+using namespace ALPAKA_ACCELERATOR_NAMESPACE;
+
+template <int NBINS, int S, int DELTA>
+struct mykernel {
+  template <typename TAcc, typename T>
+  ALPAKA_FN_ACC void operator()(const TAcc& acc, T const* __restrict__ v, uint32_t N) const {
+    ALPAKA_ASSERT_OFFLOAD(v);
+    ALPAKA_ASSERT_OFFLOAD(N == 12000);
+
+    const uint32_t threadIdxLocal(alpaka::getIdx<alpaka::Block, alpaka::Threads>(acc)[0u]);
+    if (threadIdxLocal == 0) {
+      printf("start kernel for %d data\n", N);
+    }
+
+    using Hist = HistoContainer<T, NBINS, 12000, S, uint16_t>;
+
+    auto& hist = alpaka::declareSharedVar<Hist, __COUNTER__>(acc);
+    auto& ws = alpaka::declareSharedVar<typename Hist::Counter[32], __COUNTER__>(acc);
+
+    // set off zero
+    for_each_element_in_block_strided(acc, Hist::totbins(), [&](uint32_t j) { hist.off[j] = 0; });
+    alpaka::syncBlockThreads(acc);
+
+    // set bins zero
+    for_each_element_in_block_strided(acc, Hist::totbins(), [&](uint32_t j) { hist.content[j] = 0; });
+    alpaka::syncBlockThreads(acc);
+
+    // count
+    for_each_element_in_block_strided(acc, N, [&](uint32_t j) { hist.count(acc, v[j]); });
+    alpaka::syncBlockThreads(acc);
+
+    ALPAKA_ASSERT_OFFLOAD(0 == hist.size());
+    alpaka::syncBlockThreads(acc);
+
+    // finalize
+    hist.finalize(acc, ws);
+    alpaka::syncBlockThreads(acc);
+
+    ALPAKA_ASSERT_OFFLOAD(N == hist.size());
+
+    // verify
+    for_each_element_in_block_strided(
+        acc, Hist::nbins(), [&](uint32_t j) { ALPAKA_ASSERT_OFFLOAD(hist.off[j] <= hist.off[j + 1]); });
+    alpaka::syncBlockThreads(acc);
+
+    for_each_element_in_block(acc, 32, [&](uint32_t i) {
+      ws[i] = 0;  // used by prefix scan...
+    });
+    alpaka::syncBlockThreads(acc);
+
+    // fill
+    for_each_element_in_block_strided(acc, N, [&](uint32_t j) { hist.fill(acc, v[j], j); });
+    alpaka::syncBlockThreads(acc);
+
+    ALPAKA_ASSERT_OFFLOAD(0 == hist.off[0]);
+    ALPAKA_ASSERT_OFFLOAD(N == hist.size());
+
+    // bin
+#ifndef NDEBUG
```

Comment:
Do we break anything by removing all the `#ifndef NDEBUG` ?
I don't think we ever define `NDEBUG` in CMSSW anyway.